### PR TITLE
Dependency Updates Week of Sept 18 2023

### PR DESCRIPTION
**Pull request by ujahio**
### ***Publishing a new ECR image: 3.0.15. This will require manually running docker-to-ecr.sh on the court prod account***

- Dockerfile update:
  - AWS CLI to 2.13.3

- [Successful migration run](https://app.circleci.com/pipelines/github/flexion/ef-cms/46979/workflows/f1476549-b641-4b31-9c7b-2b46307c7d6d)
 
## Notes

- Updated AWS provider to 5.17.0

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,7 +1,8 @@
 version: 2.1
 orbs:
   git-shallow-clone: guitarrapc/git-shallow-clone@2.6.0
-define: &efcms-docker-image $AWS_ACCOUNT_ID.dkr.ecr.us-east-1.amazonaws.com/ef-cms-us-east-1:3.0.14
+define:
+  efcms-docker-image: &efcms-docker-image $AWS_ACCOUNT_ID.dkr.ecr.us-east-1.amazonaws.com/ef-cms-us-east-1:3.0.15
```
Comment (jimlerza) [inline .circleci/config.yml]: this should be all on one line, without the `define`:

```
efcms-docker-image: &efcms-docker-image $AWS_ACCOUNT_ID.dkr.ecr.us-east-1.amazonaws.com/ef-cms-us-east-1:3.0.15
```

**Review by jimlerza — APPROVED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (ujahio) [reply .circleci/config.yml]: Thank you. Here's the [fix](63e2d6f)

File: package.json [hunk 2]
```diff
@@ -11,23 +11,21 @@
...
+    "@aws-sdk/client-cloudwatch": "^3.414.0",
+    "@aws-sdk/client-cognito-identity-provider": "^3.414.0",
+    "@aws-sdk/client-dynamodb": "^3.417.0",
+    "@aws-sdk/client-lambda": "^3.414.0",
+    "@aws-sdk/client-opensearch": "^3.414.0",
+    "@aws-sdk/client-route-53": "^3.414.0",
+    "@aws-sdk/client-sqs": "^3.414.0",
+    "@aws-sdk/client-ssm": "^3.414.0",
+    "@aws-sdk/credential-provider-node": "^3.414.0",
+    "@aws-sdk/lib-dynamodb": "^3.417.0",
```
Comment (jimlerza) [inline package.json]: Two packages got removed:

```
@aws-sdk/client-glue
@aws-sdk/client-s3
```

File: web-client/src/presenter/state.ts [hunk 3]
```diff
@@ -366,6 +366,7 @@ export const baseState = {
     qcSectionInProgressCount: number;
     qcIndividualInboxCount: number;
     qcIndividualInProgressCount: number;
+    unreadMessageCount?: number;
```
Comment (TomElliottFlexion) [inline web-client/src/presenter/state.ts:369]: Just curious about this

**Review by TomElliottFlexion — APPROVED** [review]

File: web-client/src/presenter/state.ts (hunk 3 above)
Comment (ujahio) [reply web-client/src/presenter/state.ts:369]: `unreadMessageCount` is set sometimes on notifications state property. Its typed as an optional field here.

**Pull request merged**